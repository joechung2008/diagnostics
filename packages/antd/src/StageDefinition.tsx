import { Table, Typography } from "antd";

const StageDefinition: React.FC<StageDefinitionProps> = ({
  stageDefinition,
}) => {
  const dataSource = Object.entries(stageDefinition).map(([key, value]) => ({
    key,
    name: key,
    value: Array.isArray(value) ? value.join(", ") : value,
  }));

  const columns = [
    {
      title: "Key",
      dataIndex: "name",
      key: "name",
    },
    {
      title: "Value",
      dataIndex: "value",
      key: "value",
    },
  ];

  return (
    <div>
      <Typography.Title id="stage-definitions-title" level={2}>
        Stage Definitions
      </Typography.Title>
      <Table
        aria-labelledby="stage-definitions-title"
        columns={columns}
        dataSource={dataSource}
        pagination={false}
      />
    </div>
  );
};

export default StageDefinition;
