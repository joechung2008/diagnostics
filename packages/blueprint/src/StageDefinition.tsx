import { HTMLTable } from "@blueprintjs/core";

const StageDefinition: React.FC<StageDefinitionProps> = ({
  stageDefinition,
}) => {
  const items = Object.entries(stageDefinition).reduce<
    KeyValuePair<string[]>[]
  >((previous, [key, value]) => [...previous, { key, value }], []);

  return (
    <div>
      <h3 className="bp6-heading">Stage Definitions</h3>
      <HTMLTable aria-label="Stage Definitions">
        <thead>
          <tr>
            <th>Key</th>
            <th>Value</th>
          </tr>
        </thead>
        <tbody>
          {items.map((item) => (
            <tr key={item.key}>
              <td>{item.key}</td>
              <td>{item.value.join(", ")}</td>
            </tr>
          ))}
        </tbody>
      </HTMLTable>
    </div>
  );
};

export default StageDefinition;
