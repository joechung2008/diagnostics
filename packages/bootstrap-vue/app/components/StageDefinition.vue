<template>
  <div>
    <h2>Stage Definitions</h2>
    <BTable :items="rows" :fields="fields" small />
  </div>
</template>

<script setup lang="ts">
import type { StageDefinitionProps } from "@/types/StageDefinitionProps";

const props = defineProps<StageDefinitionProps>();

const fields = [
  { key: "key", label: "Key" },
  { key: "value", label: "Value" }
];

const rows = computed(() =>
  Object.entries(props.stageDefinition ?? {}).map(([key, value]) => ({
    key,
    value: value.join(", ")
  }))
);
</script>
