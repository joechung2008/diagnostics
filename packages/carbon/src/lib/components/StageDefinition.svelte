<script lang="ts">
	import { DataTable } from 'carbon-components-svelte';

	const { stageDefinition }: StageDefinitionProps = $props();

	const headers = [
		{ key: 'key', value: 'Key', empty: false },
		{ key: 'value', value: 'Value', empty: false }
	];

	const rows = $derived(
		Object.entries(stageDefinition).map(([key, value]) => ({
			id: key,
			key,
			value: value.join(', ')
		}))
	);
</script>

<h2 id="stage-definitions">Stage Definition</h2>
<DataTable aria-labelledby="stage-definitions" {headers} {rows} />
