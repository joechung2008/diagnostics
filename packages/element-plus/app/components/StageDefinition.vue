<template>
    <h2 id="stage-definitions">Stage Definitions</h2>
    <el-table :data="rows" style="width: 100%" aria-labelledby="stage-definitions">
        <el-table-column prop="key" label="Key" show-overflow-tooltip />
        <el-table-column prop="value" label="Value" show-overflow-tooltip />
    </el-table>
</template>

<script setup lang="ts">
import type { StageDefinitionProps } from "@/types"

const props = defineProps<StageDefinitionProps>()

const rows = computed(() =>
    Object.entries(props.stageDefinition).map(([key, value]) => ({
        key,
        value: value.join(", ")
    }))
)
</script>
