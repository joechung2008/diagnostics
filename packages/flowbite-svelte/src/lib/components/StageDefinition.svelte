<script lang="ts">
  import {
    Heading,
    Table,
    TableBody,
    TableBodyCell,
    TableBodyRow,
    TableHead,
    TableHeadCell
  } from "flowbite-svelte";

  let { stageDefinition }: StageDefinitionProps = $props();

  const items: KeyValuePair<string[]>[] = $derived.by(() =>
    Object.entries(stageDefinition).map(([key, value]) => ({
      key,
      value
    }))
  );
</script>

<div>
  <Heading tag="h2" class="leading-loose">Stage Definitions</Heading>
  <Table aria-label="Stage Definitions">
    <TableHead>
      <TableHeadCell>Key</TableHeadCell>
      <TableHeadCell>Value</TableHeadCell>
    </TableHead>
    <TableBody>
      {#each items as item (item.key)}
        <TableBodyRow>
          <TableBodyCell>{item.key}</TableBodyCell>
          <TableBodyCell>{item.value.join(", ")}</TableBodyCell>
        </TableBodyRow>
      {/each}
    </TableBody>
  </Table>
</div>
