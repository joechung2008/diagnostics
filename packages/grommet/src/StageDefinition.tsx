import {
  Heading,
  Table,
  TableBody,
  TableCell,
  TableHeader,
  TableRow,
} from "grommet";

const StageDefinition: React.FC<StageDefinitionProps> = ({
  stageDefinition,
}) => {
  const items = Object.entries(stageDefinition).reduce<
    KeyValuePair<string[]>[]
  >((previous, [key, value]) => [...previous, { key, value }], []);

  return (
    <div>
      <Heading
        level={3}
        margin={{ bottom: "xsmall", horizontal: "none", top: "medium" }}
      >
        Stage Definitions
      </Heading>
      <Table aria-label="Stage Definitions">
        <TableHeader>
          <TableRow>
            <TableCell scope="col">Key</TableCell>
            <TableCell scope="col">Value</TableCell>
          </TableRow>
        </TableHeader>
        <TableBody>
          {items.map((item) => (
            <TableRow key={item.key}>
              <TableCell>{item.key}</TableCell>
              <TableCell>{item.value.join(", ")}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
};

export default StageDefinition;
