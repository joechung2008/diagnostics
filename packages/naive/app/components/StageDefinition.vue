<template>
  <div>
    <n-h2>Stage Definitions</n-h2>
    <n-data-table :columns="columns" :data="rows" size="small" />
  </div>
</template>

<script setup lang="ts">
import type { StageDefinitionProps } from "@/types/StageDefinitionProps"

const props = defineProps<StageDefinitionProps>()

const columns = [
  { title: "Key", key: "key" },
  { title: "Value", key: "value" }
]

const rows = computed(() =>
  Object.entries(props.stageDefinition ?? {}).map(([key, value]) => ({
    key,
    value: (value || []).join(", ")
  }))
)
</script>
