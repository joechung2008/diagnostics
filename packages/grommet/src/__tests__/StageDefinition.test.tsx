import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import StageDefinition from "../StageDefinition";

describe("StageDefinition", () => {
  it("matches snapshot", () => {
    const stageDefinition = {
      build: ["compile", "test"],
      deploy: ["upload", "restart"],
    };
    const { container } = render(
      <StageDefinition stageDefinition={stageDefinition} />
    );
    expect(container).toMatchSnapshot();
  });
});
