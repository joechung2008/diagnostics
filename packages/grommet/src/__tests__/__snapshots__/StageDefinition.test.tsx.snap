// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StageDefinition > matches snapshot 1`] = `
<div>
  <div>
    <h3
      class="StyledHeading-sc-1rdh4aw-0 hrdewH"
    >
      Stage Definitions
    </h3>
    <table
      aria-label="Stage Definitions"
      class="StyledTable-sc-1m3u5g-6 wtbPz"
    >
      <thead
        class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <th
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
            scope="col"
          >
            Key
          </th>
          <th
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 cYkJuW"
            scope="col"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="StyledTable__StyledTableBody-sc-1m3u5g-3"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
          >
            build
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
          >
            compile, test
          </td>
        </tr>
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
          >
            deploy
          </td>
          <td
            class="StyledTable__StyledTableCell-sc-1m3u5g-0 gBRlEM"
          >
            upload, restart
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
